x-pack/legacy/plugins/code/public/sagas/search.ts

```diff
@@ -146,3 +151,35 @@ function* handleReposSearchForScope(action: Action<RepositorySearchPayload>) {
 export function* watchRepoScopeSearch() {
   yield takeEvery(searchReposForScope, handleReposSearchForScope);
 }
+
+function* handleSaveSearchOptions(action: Action<SearchOptions>) {
```

Comment:
one question here is that what do we need to create new logics for handling the search scope updates? I mean, we should already have some logic handling these cases, why not adding changes in there?

Reply:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/code/public/components/search_bar/search_bar.tsx#L36 do you think it make sense to merge all the logics to handle the search scope changes together?